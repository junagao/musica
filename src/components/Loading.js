import React from 'react';

import './Loading.scss';

const Loading = () => (
  <div className="spinner-container">
    <div className="spinner">
      <span className="spinner-inner-1" />
      <span className="spinner-inner-2" />
      <span className="spinner-inner-3" />
    </div>
  </div>
);

export default Loading;
