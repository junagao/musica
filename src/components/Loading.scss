@import '../variables';

$spinner-color: $color-icon;
$spinner-background: $color-main;
$spinner-size: 30px;

.spinner-container {
  background-color: $spinner-background;
  border-radius: 0.5rem;
  border: none;
  outline: none;
  width: 54px;
  height: 54px;
}

.spinner {
  position: relative;
  top: calc(50% - #{$spinner-size / 2});
  left: calc(50% - #{$spinner-size / 2});
  width: $spinner-size;
  height: $spinner-size;
  animation: spinner 0.75s linear infinite;

  span {
    position: absolute;
    width: 100%;
    height: 100%;
    border-radius: 50%;
  }

  .spinner-inner-1 {
    background: linear-gradient(
      to right,
      rgba($spinner-color, 0) 0%,
      rgba($spinner-color, 0) 50%,
      $spinner-color 51%
    );
  }

  .spinner-inner-2 {
    background: linear-gradient(
      to top,
      rgba($spinner-background, 0) 0%,
      $spinner-background 100%
    );
  }

  .spinner-inner-3 {
    background: $spinner-background;
    $offset: $spinner-size * 0.1;
    top: $offset;
    left: $offset;
    width: $spinner-size - ($offset * 2);
    height: $spinner-size - ($offset * 2);
  }
}

@keyframes spinner {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
