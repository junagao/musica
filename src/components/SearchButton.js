import React from 'react';
import { IoIosSearch } from 'react-icons/io';

import './SearchButton.scss';

const SearchButton = () => (
  <button type="submit" className="search-button">
    <IoIosSearch />
  </button>
);

export default SearchButton;
