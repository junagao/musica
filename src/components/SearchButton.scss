@import '../variables';

.search-button {
  background-color: $color-main;
  border-radius: 0.5rem;
  border: none;
  color: $color-icon;
  cursor: pointer;
  font-size: 2.5rem;
  outline: none;
  padding: 0.4rem 0.3rem 0rem 0.3rem;
  width: 54px;
  height: 54px;
  &:hover {
    background-color: $color-darker;
  }
}
