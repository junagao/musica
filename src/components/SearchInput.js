import React from 'react';
import PropTypes from 'prop-types';

import './SearchInput.scss';

const SearchInput = ({ onSearchTermChange }) => (
  <label className="search-field" htmlFor="search">
    <input
      type="search"
      onChange={onSearchTermChange}
      name="search"
      placeholder="Search"
      className="search-input"
    />
  </label>
);

SearchInput.propTypes = {
  onSearchTermChange: PropTypes.func.isRequired,
};

export default SearchInput;
