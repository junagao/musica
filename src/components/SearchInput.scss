.search-input {
  color: #553c9a;
  border: 1px solid #553c9a;
  border-radius: 0.5rem;
  font-size: 1.5rem;
  margin-right: 0.5rem;
  padding: 0.75rem;
  width: 17rem;
  -webkit-appearance: none;
  &:focus {
    color: #553c9a;
    background-color: #faf5ff;
    border: 1px solid #553c9a;
    box-shadow: 0 1px 1px 0 #e9d8fd, 0 1px 1px 0 #d6bcfa;
    outline: none;
  }
  &::-webkit-input-placeholder {
    color: #d0bfe4;
  }
}
