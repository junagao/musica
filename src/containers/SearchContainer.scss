.search-form {
  display: flex;
  align-items: center;
  margin-right: 0.75rem;
  height: 54px;
}
