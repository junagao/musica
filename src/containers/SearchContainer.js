import React from 'react';
import PropTypes from 'prop-types';

import SearchButton from '../components/SearchButton';
import SearchInput from '../components/SearchInput';

class SearchContainer extends React.Component {
  constructor() {
    super();
    this.state = {
      searchTerm: '',
    };
  }

  onSearchTermChange = (e) => {
    this.setState({ searchTerm: e.target.value });
  };

  handleSubmit = (e) => {
    const { onSearch } = this.props;
    const { searchTerm } = this.state;
    e.preventDefault();
    onSearch(searchTerm);
    e.currentTarget.reset();
  };

  render() {
    return (
      <form className="search-form" onSubmit={this.handleSubmit}>
        <SearchInput onSearchTermChange={this.onSearchTermChange} />
        <SearchButton />
      </form>
    );
  }
}

SearchContainer.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchContainer;
