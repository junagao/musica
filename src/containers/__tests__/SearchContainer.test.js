import React from 'react';
import { shallow } from 'enzyme';

import SearchContainer from 'containers/SearchContainer';
import SearchInput from 'components/SearchInput';
import Loading from 'components/Loading';
import SearchButton from 'components/SearchButton';

describe('SearchContainer', () => {
  let wrapper;
  const onSearchMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<SearchContainer onSearch={onSearchMock} loading />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a form element with className `search-form`', () => {
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render one SearchInput component', () => {
    expect(wrapper.find(SearchInput).length).toEqual(1);
  });

  it('should render one Loading component if this.props.loading is true', () => {
    expect(wrapper.find(Loading).length).toEqual(1);
  });

  it('should render one Loading component if this.props.loading is false', () => {
    wrapper.setProps({ loading: false });
    expect(wrapper.find(SearchButton).length).toEqual(1);
  });
});
