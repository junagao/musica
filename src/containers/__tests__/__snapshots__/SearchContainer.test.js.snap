// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should render correctly given the required props 1`] = `
<form
  className="search-form"
  onSubmit={[Function]}
>
  <SearchInput
    onSearchTermChange={[Function]}
  />
  <Loading />
</form>
`;
